app/invocation/invocation.tsx

```diff
@@ -262,16 +263,19 @@ export default class InvocationComponent extends React.Component<Props, State> {
   }
 
   scheduleRefetch() {
-    clearTimeout(this.timeoutRef);
-    // Refetch invocation data in 3 seconds to update status.
-    this.timeoutRef = window.setTimeout(async () => {
-      // Before fetching the invocation, wait for the runner execution to be
-      // fetched, so we don't keep canceling the execution fetch if it takes
-      // longer than the invocation poll interval.
-      if (this.runnerExecutionRPC) await this.runnerExecutionRPC;
-
-      this.fetchInvocation();
-    }, 3000);
+    if (!this.refetchInProgress) {
+      this.refetchInProgress = true;
```

Comment:
nit: to avoid introducing more component state (`this.refetchInProgress`) we could check whether there's an existing timeout and avoid starting a new one if one already exists

```
scheduleRefetch() {
  this.timeoutRef ??= setTimeout(async () => {
    /* ... */
    this.timeoutRef = null;
  }, 3000);
}
```